src/app/model/Knight.ts

```diff
@@ -0,0 +1,63 @@
+import { Board } from './Board';
+import { Piece } from './Piece';
+import { BOARDSIZE } from './Constances';
+import { PieceMovement } from './PieceMovement';
+import { columns, Row, Side, Square } from './Types';
+
+export class Knight extends Piece implements PieceMovement {
+  constructor(id: string, side: Side) {
+    super('uselessId', side);
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo = [] as Square[];
+    movesToGo = movesToGo.concat(this.movesOnEmptyBoard(position, board));
+    // TODO: usuń pola zajęte
+    movesToGo = this.removeImpossibleSquares(movesToGo, board);
+    return movesToGo;
+  }
```

Comment:
To z pewnością można zapisać tak :) : 
```suggestion
  possibleMoves(position: Square, board: Board): Square[] {
    return removeImpossibleSquares(this.movesOnEmptyBoard(position, board))
  }
```
Albo wtedy remove nazwac withoutSquaresWithAllyPieces np :) ?

Reply:
done